### PR TITLE
Backend for Filtering Projects

### PR DESCRIPTION
**Pull request by Vishalk91-4**
Partially Fixes issue #3

Created **getAllProjects go** file, so as to fetch all projects and display them, this is a subpart of **search bar**, which will be used to search both orgs or projects directly

image

@EshaanAgg could you please review this pull request

### CODE REVIEW

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
@Vishalk91-4 The PR looks good, but there is no real need for this right now. Instead, I would advise you to change this endpoint to accept a query parameter called `filterText`, and then this endpoint returns all the projects whose name or description matches the same. Then this endpoint could be used in the frontend directly, and we wouldn't require searching on frontend.

**Comment by Vishalk91-4** [comment]
@EshaanAgg, could you please review it
I just checked that main.go has filtering of projects by year also, do we have to keep it separate

**Comment by EshaanAgg** [comment]
@Vishalk91-4 Good job, and the PR now performs filtering. But instead of creating a new endpoint, it would be better to change the `/projects` endpoint to accept an optional query parameter called `filterText`. If the same is provided, then we use the `like` query, else if the same is not provided or is `""`, we provide all the organizations. It should be an easy fix. 

We will keep by-year filtering separate as it is being used in the graph component for UI.

**Comment by Vishalk91-4** [comment]
@EshaanAgg, I've added a commit, could you check

File: api/getProjectsByFilter.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	filterText := c.DefaultQuery("filterText", "")
+	client := handlers.New()
+	defer client.Close()
+
+	var projects []database.ProjectThumbnail
+	var err error
+
+	if filterText != "" {
+
```
Comment (EshaanAgg) [inline api/getProjectsByFilter.go:21]: Remove extra newline.

File: api/getProjectsByFilter.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		projects, err = client.GetProjectsByFilter(filterText)
+	} else {
+
```
Comment (EshaanAgg) [inline api/getProjectsByFilter.go:24]: Remove extra newline.

File: database/handlers/project.go [hunk 3]
```diff
@@ -8,6 +8,31 @@ import (
 	"github.com/lib/pq"
 )
 
+func (client Client) GetProjectsByFilter(filterText string) ([]database.ProjectThumbnail, error) {
+	rowsRs, err := client.Query("SELECT id, name, description, programYear, programTerm FROM projects WHERE name LIKE '%' || $1 OR description LIKE '%' || $1;", filterText)
```
Comment (EshaanAgg) [inline database/handlers/project.go:12]: Can't understand the like clauses in this query. Shouldn't it be like `name LIKE '%$1%' OR description LIKE '%$1$'`?

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
Need some changes.

File: api/getProjectsByFilter.go (hunk 1 above)
Comment (Vishalk91-4) [reply api/getProjectsByFilter.go]: Removed

**Pull request merged**